controllers/propagator/propagation.go

```diff
@@ -225,14 +225,11 @@ func (r *PolicyReconciler) handleDecisions(
 	failedClusters = map[string]bool{}
 
 	for _, pb := range pbList.Items {
-		subjects := pb.Subjects
-		for _, subject := range subjects {
-			if !(subject.APIGroup == policiesv1.SchemeGroupVersion.Group &&
-				subject.Kind == policiesv1.Kind &&
-				subject.Name == instance.GetName()) {
-				continue
-			}
-
+		// Only handle the first matching subject in pb.spec.subjects
+		// pb.spec.subjects is a required field, it will never be empty
+		// no need to do null check
+		subject := pb.Subjects[0]
```

Comment:
I couldn't confirm if a required field could be provided but empty... eg an empty string or a list with 0 items. If you've tested it, I'll take your word for it :)

Reply:
You are right. It will be bypassed if you pass in an empty array. So I also added minLength to ensure it contains 1 item.